classes/migration/upgrade/v3_5_0/I7135_CreateNewRorRegistryCacheTables.php

```diff
@@ -0,0 +1,171 @@
+            ->join('ror_settings as r_s',
+                function (JoinClause $join) {
+                    $join
+                        ->on('r.ror_id', '=', 'r_s.ror_id')
+                        ->where('r_s.setting_name', '=', 'name');
+                }
+            )
+            ->join('author_settings as a_s',
+                function (JoinClause $join) {
+                    $join
+                        ->on('r_s.setting_value', '=', 'a_s.setting_value')
+                        ->on('r_s.locale', '=', 'a_s.locale')
+                        ->where('a_s.setting_name', '=', 'affiliation');
+                }
+            )
+            ->leftJoin('author_affiliations as aa', 'a_s.author_id', '=', 'aa.author_id')
+            ->where('aa.author_id', '=', null)
+            ->select(['r.ror_id', 'a_s.author_id'])
+            ->distinct()
+            ->get();
+
+        foreach ($rows as $row) {
+            $ror = Repo::ror()->get($row->ror_id);
+
+            $affiliation = Repo::affiliation()->newDataObject();
+            $params = [
+                "id" => null,
+                "authorId" => $row->author_id,
+                "ror" => $ror->_data['ror'],
+                "name" => $ror->_data['name']
+            ];
+            $affiliation->setAllData($params);
+
+            Repo::affiliation()->dao->updateOrInsert($affiliation);
+        }
+    }
+
+    /**
+     * Migrates affiliations which have not migrated yet.
+     * Only migrate rows which don't exist.
+     *
+     * select distinct a.author_id, a_s.locale, a_s.setting_value
+     * from authors as a
+     * join author_settings as a_s on a.author_id = a_s.author_id and a_s.setting_name = 'affiliation'
+     * left join author_affiliations as aa on a.author_id = aa.author_id
+     * where aa.author_id is null
+     *
+     * @return void
+     */
+    public function migrateNonRorAffiliations(): void
+    {
+        $rows = DB::table('authors as a')
+            ->join('author_settings as a_s',
+                function (JoinClause $join) {
+                    $join
+                        ->on('a.author_id', '=', 'a_s.author_id')
+                        ->where('a_s.setting_name', '=', 'affiliation');
+                }
+            )
+            ->leftJoin('author_affiliations as aa', 'a.author_id', '=', 'aa.author_id')
+            ->where('aa.author_id', '=', null)
+            ->select(['a.author_id', 'a_s.locale', 'a_s.setting_value'])
+            ->distinct()
+            ->get();
+
+        foreach ($rows as $row) {
+            $affiliation = Repo::affiliation()->newDataObject();
+            $params = [
+                "id" => null,
+                "authorId" => $row->author_id,
+                "ror" => '',
+                "name" => [
+                    $row->locale => $row->setting_value
+                ]
+            ];
+            $affiliation->setAllData($params);
+
+            Repo::affiliation()->dao->updateOrInsert($affiliation);
```

Comment:
Here as well, we should use the query builder to insert or update the affiliation